google/async_resumable_media/requests/download.py

```diff
@@ -87,45 +76,37 @@ async def _write_to_stream(self, response):
             ~google.resumable_media.common.DataCorruption: If the download's
                 checksum doesn't agree with server-computed checksum.
         """
-        expected_md5_hash = _get_expected_md5(
-            response, self._get_headers, self.media_url
+
+        expected_checksum, checksum_object = sync_helpers._get_expected_checksum(
+            response, self._get_headers, self.media_url, checksum_type=self.checksum
         )
 
-        if expected_md5_hash is None:
-            md5_hash = _DoNothingHash()
-        else:
-            md5_hash = hashlib.md5()
-
-        '''
-        while True:
-            breakpoint()
-            chunk = await response.content.read(1)
-            local_hash = _add_decoder(response, md5_hash)
-            if chunk == b'':
-                break
-            self._stream.write(chunk)
-            local_hash.update(chunk)
-        '''
+        local_checksum_object = _add_decoder(response, checksum_object)
 
-        local_hash = _add_decoder(response, md5_hash)     
-        async for chunk in response.content.iter_chunked(_helpers._SINGLE_GET_CHUNK_SIZE):
+        async for chunk in response.content.iter_chunked(_request_helpers._SINGLE_GET_CHUNK_SIZE):
             self._stream.write(chunk)
-            local_hash.update(chunk)
+            local_checksum_object.update(chunk)
 
-        if expected_md5_hash is None:
+        if expected_checksum is None:
             return
 
-        actual_md5_hash = base64.b64encode(md5_hash.digest())
-        # NOTE: ``b64encode`` returns ``bytes``, but ``expected_md5_hash``
-        #       came from a header, so it will be ``str``.
-        actual_md5_hash = actual_md5_hash.decode(u"utf-8")
-        if actual_md5_hash != expected_md5_hash:
-            msg = _CHECKSUM_MISMATCH.format(
-                self.media_url, expected_md5_hash, actual_md5_hash
-            )
-            raise common.DataCorruption(response, msg)
-
-    async def consume(self, transport):
+        else:
+            actual_checksum = sync_helpers.prepare_checksum_digest(checksum_object.digest())
+            if actual_checksum != expected_checksum:
+                msg = _CHECKSUM_MISMATCH.format(
+                    self.media_url,
+                    expected_checksum,
+                    actual_checksum,
+                    checksum_type=self.checksum.upper(),
+                )
+                raise common.DataCorruption(response, msg)
+
+    async def consume(
+        self,
+        transport,
+        timeout=_request_helpers._DEFAULT_CONNECT_TIMEOUT
+    ):
+        # TODO(timeout tuple incompatibility in aiohttp, so singular timeout value)
```

Comment:
please expand this. also format is typically TODO(name): message. 

Feel free to either put your github name, or yours and mine

Reply:
Done, changed format of the timeout and added more.